### ISSUE TITLE
bug in pairwise_euclidean_distance

### ISSUE THREAD

**Issue by walternat1ve** [Lightning-AI/metrics#1351]
https://github.com/Lightning-AI/metrics/blob/e1c3fda24f90367803c2b04315ad7c8bced719db/torchmetrics/functional/pairwise/euclidean.py#L34
this line can become negative, resulting in a failure with the sqrt function and thus return "nan"

you can test this easily by checking this code:

`pairwise_euclidean_distance(torch.tensor([[772., 112.], [772.20001, 112.], [772.20001, 112.], [772., 112.00000], [772.2, 112.00000], [772.0, 112.00000], [772.01, 112.00000], [772.00000000000001, 112.00000], [772.000001, 112.00000], [772.00001, 112.00000], [772.0001, 112.00000], [772.001, 112.00000], [772.01, 112.00000], [772.99, 112.00000]], dtype=torch.float32))`

**Comment by walternat1ve** [comment]
furthermore i noticed there is also some noticeable difference in the values it computes. compared to many other similar functions, which among themselves are consistent in the output. please check those as well.

**Comment by SkafteNicki** [comment]
Hi @walternat1ve, I created PR to take care of the precision issue you mention first. For your second comment do you have any examples? We are comparing towards sklearn, so in most cases it should be correct however there may be corner cases we have not thought about.

**Comment by walternat1ve** [comment]
here is one example, comparing various implementations:

inputs:
```
labels = tensor([[709.79999, 192.10001]])
detections = tensor([[ 709.,  192.],
        [ 282.,  127.],
        [ 311.,  106.],
        [1178.,   13.]])
```

4 different functions:
```
(labels - detections).pow(2).sum(1).sqrt()
Out[1]: tensor([  0.80621, 432.72491, 407.98856, 501.28641])

from torchmetrics.functional import pairwise_euclidean_distance
pairwise_euclidean_distance(labels, detections)
Out[2]: tensor([[  0.86603, 432.72482, 407.98843, 501.28647]])

import torch
pdist = torch.nn.PairwiseDistance()
pdist(labels, detections)
Out[3]: tensor([  0.80622, 432.72491, 407.98856, 501.28641])

from sklearn.metrics.pairwise import euclidean_distances
euclidean_distances(labels, detections)
Out[4]: array([[    0.80621,      432.72,      407.99,      501.29]], dtype=float32)
```

**Comment by SkafteNicki** [comment]
Thanks for the further examples.
Happy to report with the changes in PR #1352 that the metric now also seems to match sklearns implementation on the provided example:
![image](https://user-images.githubusercontent.com/24896311/203486807-e6456bf9-05d4-4189-a91c-9f5cc6addb02.png)
